### ISSUE TITLE
options.template not working

### ISSUE THREAD

**Issue by Redsandro** [jsdoc2md/grunt-jsdoc-to-markdown#11]
It seems that if I specify a template for the Grunt task, the __filename string__ will be the **content** of the output file.

E.g.:

```js
jsdoc2md        : {
    startpage       : {
        options         : {
            template        : 'docs/partials/start.hbs',
            helper          : 'docs/helpers/*.js',
            partial         : 'docs/partials/*.hbs'
        },
        src             : 'app/*.js',
        dest            : 'app/README.md'
    }
}
```

`README.md`:
> docs/partials/start.hbs

When removing `tempate` from the Grunt task, it works fine. Except ofcourse that my template isn't used.

`--template` in `jsdoc2md` command line works fine.

**Comment by 75lb** [comment]
hi.. you supply the `template` as a string - everything is correct, please [read the docs](https://github.com/jsdoc2md/jsdoc-to-markdown/blob/master/docs/API.md#jsdoc2mdrenderoptions--promise).

It's impractical to supply the template as a string to the CLI, so it accepts a filename instead. Sorry if it's confusing.

**Comment by Redsandro** [comment]
Hi @75lb ah I get it. I did actually read the docs. I've been reading everything the last few days to get things right. And this part it's anti-intuitive. I do supply the template as a string. `'docs/partials/start.hbs'` is both a template and a string. See the confusion?

I know now that I misunderstand `data` too. But I assumed (_after_ reading the docs) that `template` would be a template file, and `data` (raw template data) meant a string of the contents of a template file (what `template` actually is - no idea what that makes `data`).

Before I complain, I want you to know that this project is really cool. It's quite easy to set up as compared to others, and it even detects my custom tags. Well done. :+1: 

That said, `template` needing to be a complete template content string is just silly. I have this big project with many documentation files where the jsdoc writeouts are just small parts. Now I have to load all these huge templates in memory of the Grunt file.

Grunt is the build, watch and automation tool. Now pretty much the entire (templated) documentation needs to be in memory permanently even though I only need the documentation functions occasionally.

I have looked over the source code to make a PR but I don't really understand how this is mapped. Can I ask/suggest you add a small fix function between `grunt-jsdoc-to-markdown` and `jsdoc-to-markdown`?

```js
if (fs.existsSync(template.split("\n")[0])) // Don't pass a megabyte worth of data - who knows what would happen
    template = fs.readFileSync(template);
```

This way filenames are accepted. More intuitive and similar to `jsdoc2md cli`. And it's backwards compatible. No one will notice.
